client/pkg/tlsutil/versions.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tlsutil
+
+import (
+	"crypto/tls"
+	"fmt"
+)
+
+// tlsVersions is a map of TLS version string to the value of tls.Config.Min/MaxVersion.
+var tlsVersions = map[string]uint16{
```

Comment:
Since etcd only supports TLS 1.2 pre-PR, does it make sense to add support for obsolete, insecure and disabled-by-default TLS 1.0 and TLS 1.1? Enabling TLS 1.3 would be sufficient in my opinion, and configuration would then allow a meaningful choice between forcing TLS 1.2 only or TLS 1.3 only in addition to default behavior of enabling both.